packages/picasso.js/src/core/chart-components/axis/axis.js

```diff
@@ -48,6 +52,9 @@ function resolveLocalSettings({
   localStgns.align = resolveAlign(settings.settings.align, dock);
   localStgns.labels.tiltAngle = Math.max(-90, Math.min(localStgns.labels.tiltAngle, 90));
 
+  const { paddingStart } = localStgns;
+  localStgns.paddingStart = typeof paddingStart === 'function' ? paddingStart.call(null, chart, state) : paddingStart;
```

Comment:
exposing `state` which is an internal variable containing tons of stuff is a big nono - don't expose more than you need for the job. Typically the first param is an object containing the minimum amount of properties needed for the intended purpose, that way we reduce the risk of breaking changes if we need more things.

Reply:
The parameters passed to the callback function should ideally also follow a similar pattern across all components. Typically it looks like this:
```js
paddingStart: ({
  resources: { scale, formatter }, // scale and formatter is the same function exposed by the chart instance
  /*  additional properties that may be included here such as the scale used by the component and/or the dataset and/or a datum */
}) => {}
```

In your case I believe it could be enough with the above example.

Reply:
Ok. I only have to reach `state.innerRect` and the major `scale`. Feels wrong to send that as a specific parameter (`innerRect`)?

Reply:
At the time of resolving the settings. The size of the component hasn't be fully resolved yet. So the "rect" that you could pass down as a parameter wouldn't be accurate in all cases.

You best bet here is to use closure to access a guess-estimated size.